File: Basic/Functions/219432L.go
package main

import (
	"fmt"
)

func solution() {

	var a, b int
	fmt.Scan(&a, &b)

	func() {
		if a < b {
			a, b = b, a
		}
		for b != 0 {
			a %= b
			if a < b {
				a, b = b, a
			}
		}
	}()

	fmt.Println(a)
}

func main() {
	TEST_CASE := !true
	t := 1
	if TEST_CASE {
		fmt.Scanf("%d", &t)
	}

	for t != 0 {
		solution()
		t--
	}
}

File: Basic/Functions/219432L.c
#include <stdio.h>
#define intll long long

void swap(int *a, int *b) {
	int temp = a;
	a = b;
	b = temp;
}

int gcd(int a, int b) {
	if (a < b) {
		swap(a, b);
	}
	while (b != 0) {
		a %= b;
		if (a < b) {
			swap(a, b);
		}
	}
	return a;
}

void solution() {

	int a, b;
	scanf("%d %d", &a, &b);

	printf("%d\n", gcd(a, b));
}

int main() {
	char TEST_CASE = 0;
	int t = 1;
	if (TEST_CASE == 1) {
		scanf("%d", &t);
	}

	while (t--) {
		solution();
	}
	return 0;
}